avm/res/synapse/workspace/main.bicep

```
@@ -246,51 +246,57 @@ module synapse_integrationRuntimes 'integration-runtime/main.bicep' = [
 
 // Workspace encryption with customer managed keys
 // - Assign Synapse Workspace MSI access to encryption key
-module workspace_cmk_rbac 'modules/nested_cmkRbac.bicep' =
-  if (encryptionActivateWorkspace) {
-    name: '${workspace.name}-cmk-rbac'
-    params: {
-      workspaceIndentityPrincipalId: workspace.identity.principalId
-      keyvaultName: !empty(customerManagedKey.?keyVaultResourceId) ? cMKKeyVault.name : ''
-      usesRbacAuthorization: !empty(customerManagedKey.?keyVaultResourceId)
-        ? cMKKeyVault.properties.enableRbacAuthorization
-        : true
-    }
-    scope: encryptionActivateWorkspace
-      ? resourceGroup(
-          split((customerManagedKey.?keyVaultResourceId ?? '//'), '/')[2],
-          split((customerManagedKey.?keyVaultResourceId ?? '////'), '/')[4]
-        )
-      : resourceGroup()
+module workspace_cmk_rbac 'modules/nested_cmkRbac.bicep' = if (encryptionActivateWorkspace) {
+  name: '${workspace.name}-cmk-rbac'
+  params: {
+    workspaceIndentityPrincipalId: workspace.identity.principalId
+    keyvaultName: !empty(customerManagedKey.?keyVaultResourceId) ? cMKKeyVault.name : ''
+    usesRbacAuthorization: !empty(customerManagedKey.?keyVaultResourceId)
+      ? cMKKeyVault.properties.enableRbacAuthorization
+      : true
   }
+  scope: encryptionActivateWorkspace
+    ? resourceGroup(
+        split((customerManagedKey.?keyVaultResourceId ?? '//'), '/')[2],
+        split((customerManagedKey.?keyVaultResourceId ?? '////'), '/')[4]
+      )
+    : resourceGroup()
+}
 
 // - Workspace encryption - Activate Workspace
-module workspace_key 'key/main.bicep' =
-  if (encryptionActivateWorkspace) {
-    name: '${workspace.name}-cmk-activation'
-    params: {
-      name: customerManagedKey!.keyName
-      isActiveCMK: true
-      keyVaultResourceId: cMKKeyVault.id
-      workspaceName: workspace.name
-    }
-    dependsOn: [
-      workspace_cmk_rbac
-    ]
+module workspace_key 'key/main.bicep' = if (encryptionActivateWorkspace) {
+  name: '${workspace.name}-cmk-activation'
+  params: {
+    name: customerManagedKey!.keyName
+    isActiveCMK: true
+    keyVaultResourceId: cMKKeyVault.id
+    workspaceName: workspace.name
   }
+  dependsOn: [
+    workspace_cmk_rbac
+  ]
+}
+
+// - Workspace Entra ID Administrator
+module workspace_administrator 'administrators/main.bicep' = if (!empty(administrator.login)) {
+  name: '${workspace.name}-administrator'
+  params: {
+    workspaceName: workspace.name
+    administrator: administrator
+  }
+}
```

As per the above suggestion this could change to:

```suggestion
module workspace_administrator 'administrators/main.bicep' = if (!empty(administrator)) {
  name: '${workspace.name}-administrator'
  params: {
    workspaceName: workspace.name
    administrator: administrator
  }
}
```